apps/docs/src/examples/misc/gaussian-splatting/LumaSplatsThree/LumaSplatsThree.svelte

```diff
@@ -57,18 +62,21 @@
     { autoStart: false }
   )
 
-  let previousEnvironment: Scene['environment']
-  let previousBackground: Scene['background']
-  let previousBackgroundBluriness: Scene['backgroundBlurriness']
-  $: if ($splats && $splats[1]) {
-    previousEnvironment = scene.environment
-    previousBackground = scene.background
-    previousBackgroundBluriness = scene.backgroundBlurriness
-    scene.environment = $splats[1]
-    scene.background = $splats[1]
-    scene.backgroundBlurriness = 0.5
-    invalidate()
-  }
+  let previousEnvironment: Scene['environment'] = $state(null)
+  let previousBackground: Scene['background'] = $state(null)
+  let previousBackgroundBluriness: Scene['backgroundBlurriness'] = $state()
+
+  $effect(() => {
+    if ($splats && $splats[1]) {
+      previousEnvironment = scene.environment
+      previousBackground = scene.background
+      previousBackgroundBluriness = scene.backgroundBlurriness
+      scene.environment = $splats[1]
+      scene.background = $splats[1]
+      scene.backgroundBlurriness = 0.5
+      invalidate()
+    }
+  })
```

Comment:
this should probably be broken up into multiple `$derived`s and `$effect`s but we can do that in a later pr somewhere else.